src/main/php/PDepend/Util/Utf8Util.php

```diff
@@ -54,15 +54,14 @@ final class Utf8Util
 {
     public static function ensureEncoding($raw)
     {
+        $encoding = 'UTF8';
         if (function_exists('mb_detect_encoding')) {
-            $encoding = mb_detect_encoding($raw);
-        } else {
-            $encoding = 'UTF8';
+            $encoding = mb_detect_encoding($raw) ?: $encoding;
         }
```

Comment:
This is a logic change, this could previously be `false` (which would fallback to `'ASCII'` equivalent when used in `iconv()`) so we're actually changing the default from `'ASCII'` to `'UTF8'`. It's not a big change and looks fine regarding the  `Utf8Util::ensureEncoding` but it should be noted in the PR description as a reminder.

It could be written like this to assign the variable only once:
```suggestion
        $encoding = (function_exists('mb_detect_encoding') ? mb_detect_encoding($raw) : null) ?: 'UTF8';
```

But double-ternary may not please the rest of the team. :) So it's as you wish to take or not.

Reply:
I'm not a fan of double-ternary either :)
The interpreter likely will produce the same machine code anyway.

Reply:
I'm pretty sure the interpreter does not cleanup for you such double assignation.

Reply:
Arh, didn't see you use of a fallback `null`, yeah this woudn't compile to the same opcode.

Reply:
The `null` is here to avoid the `'UTF8'` redundancy. But what I meant is the following cannot be reduced by the compiler optimizer:
```php
$foo = 'bar'; // memory variable written once
$foo = call() ?: $foo; // memory variable written twice
```
But that's obviously not a blocking point. We have way bigger bottleneck that those microscopic ones.

The important point above is `$encoding` could previously be `false`. Then it no longer can.

This is the one thing that should be mentioned in the PR description.